SPECIFICATION.md

```diff
@@ -41,13 +41,28 @@ Specification
 ### Creating a Client
 A developer will create a client object with some configuration values. This configuration:
 
-- Must support API credentials (key and secret) for authentication.
-- Must support OAuth1 token for authentication.
-- Must require an authentication method (credentials or oauth) for construction.
+- Must support multiple authentication credentials (see '[]Authenticating Requests](authenticating-requests)')
+- Must require at least one authentication credential for construction.
 - Must allow override of the base URL(s) (currently there are multiple base urls).
-- Must allow a shared secret (for WebHook request signatures).
 - Must not force a singleton (but may provide access to a singleton).
  
+### Authenticating Requests
+The current state of the API allows multiple authentication credentials. The client library:
+
+- Must allow a combination (one or more) of the following credential types:
+    - Must allow an API Key and Secret (current API)
+    - Must allow a signature secret (use for _some_ API requests, and to validate WebHook signatures)
+    - Must allow multiple private keys for JWT authentication (replacing key and secret)
+    - May allow OAuth1 token for authentication
+- Must not allow more than one credential per type.
```

Comment:
Sure, point here is while you could create a client instance with both API key / secret _and_ a private key, you shouldn't be able to create one with two private keys. 

Would this work?
> Must not allow multiple credentials of the same type (two private keys, two sets of key and secret, etc).